webknossos-tracingstore/proto/geometry.proto

```diff
@@ -21,3 +21,8 @@ message BoundingBox {
     required int32 height = 3;
     required int32 depth = 4;
 }
+
+message NamedBoundingBox {
+  optional string name = 1;
+  required BoundingBox boundingBox = 2;
```

Comment:
Awesome! What do you think about an additional `isVisible` parameter so that we can persist when the user disabled such a box?

Reply:
Sounds good :D  Do you think we should also give them IDs? so that when the user edits them we have an identifier? The problem of course being backwards compatibility (the ones already saved in `userBoundingBox` don’t have IDs (though we could assign 0 as special case for them or something)

Reply:
Hmm, not sure whether the ids are really necessary. A tracing probably won't have more than a few bounding boxes (and 640K ought to be enough for anybody). So, just sending all bounding boxes on an update doesn't seem like a big deal to me. Optimizing this for performance could be done later. Or do you think that we should do this right away to avoid another breaking change?

Btw, the ID for `userBoundingBox` could simply be `userBoundingBox`. Or do you think it should be unique across all tracings? If so, creating a new boundingbox needs to go through the back-end which would be another pain, I guess.

Reply:
No need for this to be unique across all tracings, that’s fine.
Yeah I’m just a bit scared that once this is available, people will start creating NMLs with thousands of bboxes :D If they become part of the updateTracing update action then they are also included in every move/zoom/visibility update. But maybe that’s okay since those happen only every 30s… :thinking:

Reply:
by the way: do you think they should have a color attribute also?

Reply:
> Yeah I’m just a bit scared that once this is available, people will start creating NMLs with thousands of bboxes :D

Yeah, very valid point. Maybe we should artificially limit this to 10 bounding boxes? 

> by the way: do you think they should have a color attribute also?

This feature grows and grows slowly :see_no_evil: Would be nice, probably. But also not essential in my opinion. Auto-generating the colors should be fine, in my opinion :thinking: 

Btw, could this feature be unified with the general mesh support? Might be useful, but I'm not sure how well this can be done.

Reply:
I talked to @youri-k about this (in fact, I’m mostly yielding this project to them for now). We agreed that it would be better to build this in a scalable way right away, rather than creating artificial limits first which we then have to “fix” in a backwards-compatible way once users need more than 10.
So we’d suggest introducing ids for the bboxes (like for trees) and new create/delete/change update actions for them. I’d say the original optional user bbox should then not be saved again but moved to that list by the frontend (where it can also get new attributes like a name). We certainly haven’t thought about all cases but I think this is the direction in which to think.
I agree that we can defer colors for now.
I don’t understand how this relates to mesh support, could you elaborate?

Reply:
Sounds good :+1:

> I don’t understand how this relates to mesh support, could you elaborate?

A tracing can contain multiple "entities": skeletons, comments, meshes, isosurfaces and now bounding boxes (some are persisted in the NML, some a only persisted in the db (meshes) and some are never persisted (isosurfaces)). I'm not sure how well our design unifies these different concepts. A common interface (where possible) might make sense. In the end, I mostly care about the UI, since that should be well-structured. And for the user, a bounding box and a mesh might be so similar that the UI should present them in the same place (which might require that we can unify these in code, as well).
Sorry, I'm rambling a bit. I don't have a clear picture of this yet, but I feel like we should think about how the different concepts can be designed in a coherent hierarchy (?).